src/core/tasks.py

```diff
@@ -0,0 +1,140 @@
+    )
+    new_csv.close()
+
+    with open('new.csv', 'rb') as f:
+        reader = csv.reader(
+            f.read().splitlines()
+        )
+        return reader
+
+
+@task(name='add-metadata')
+def add_metadata():
+    """Adds book metadata from CSVs on an FTP site.
+
+    Accesses a specified FTP dir, copies CSVs to
+    the local dir then deletes them from FTP, reads
+    the CSV and matches ISBN13s to Rua books and
+    updates their metadata with the matching CSV
+    column values, and sends a notification email.
+    """
+    if settings.FTP_METADATA_FEATURE:
+        service = ServiceHandler(service=JuraUpdateService)
+
+        ftp = ftplib.FTP(getattr(settings, 'FTP_URL', None))
+        ftp.login(
+            getattr(settings, 'FTP_USERNAME', None),
+            getattr(settings, 'FTP_PASSWORD', None)
+        )
+        ftp.cwd(getattr(settings, 'FTP_FOLDER', None))
+        files = []
+
+        try:
+            files = ftp.nlst()
+        except ftplib.error_perm, resp:
+            if str(resp) == '550 No files found':
+                print 'No files in this directory'
+            else:
+                raise
+
+        for f in files:
+            if '.csv' in f:
+                ftp.retrbinary(
+                    'RETR ' + f,
+                    open(f, 'wb').write
+                )
+                ftp.delete(f)
+                ftp.close()
+
+                _csv = _read_csv(f)
+                headers = next(_csv)
+                isbn_index = headers.index('ISBN13')
+                bisac_index = headers.index('Bisac Code(s)')
+                description_index = headers.index('Long Description')
+                data = [row for row in _csv]
+                isbns_processed = []
+
+                for row in data:
+                    isbn = row[isbn_index]
+                    if '-' not in isbn: # Add hyphens to ISBNs to match UCP Rua format.
+                        isbn = '-'.join(
+                            [
+                                isbn[:3],
+                                isbn[3],
+                                isbn[4:7],
+                                isbn[7:12],
+                                isbn[12]
+                            ]
+                        )
+                    bisacs = row[bisac_index]
+                    description = row[description_index]
+                    identifier = models.Identifier.objects.filter(value=isbn).first()
+
+                    if identifier and isbn not in isbns_processed:
+                        book = identifier.book
+
+                        for subj in book.subject.all():
+                            book.subject.remove(subj)
+
+                        for bisac in bisacs.split():
+                            bisac_lookup = [
+                                row for row in _read_csv('bisac.csv') if bisac in row
+                            ]
+                            new_subject, created = models.Subject.objects.get_or_create(
```

Comment:
```created``` is unused, thus seems to me you can reduce confusion by doing this

```new_subject, _ = models.Subject.objects.get_or_create(...)``` with the ```_``` indicating your not using it